lib/active_model/serializer.rb

```diff
@@ -379,6 +379,7 @@ def associations(include_directive = ActiveModelSerializers.default_include_dire
     def serializable_hash(adapter_options = nil, options = {}, adapter_instance = self.class.serialization_adapter_instance)
       adapter_options ||= {}
       options[:include_directive] ||= ActiveModel::Serializer.include_directive_from_options(adapter_options)
+      options[:fields] ||= adapter_options[:fieldset] && adapter_options[:fieldset].fields_for(json_key)
```

Comment:
This seems to duplicate logic in the adapter.

```
def fields
   fieldset && fieldset.fields_for(serializer.json_key)
end
```

which should it be?

Is the problem here that the adapter does `options.delete(:fields)`?